src/tests/ftest/launch.py

```diff
@@ -20,164 +20,471 @@
-    print("\tpoolevict --run all client pool eviction related tests")
-    print("\tpoolinfo --run all pool info retrieval related tests")
-    print("\tquick --run tests that complete quickly, with minimal resources ")
-    print("\n")
-    print("You can also specify the sparse flag -s to limit output to "
-          "pass/fail.")
-    print("Example command: launch.py -s pool")
-    print("\n")
-    exit()
-
-def run_test(_file, use_tags=True):
+    yaml_data = {}
+    if os.path.isfile(yaml_file):
+        with open(yaml_file, "r") as open_file:
+            try:
+                file_data = open_file.read()
+                yaml_data = yaml.safe_load(file_data.replace("!mux", ""))
+            except yaml.YAMLError as error:
+                display("Error reading {}: {}".format(yaml_file, error))
+                exit(1)
+    return yaml_data
+
+
+def clean_test_environment(machines, debug):
+    """Clean the daos test environment on each specified machine.
+
+    Args:
+        machines (list): a list of machines to clean
+        debug (bool): whether or not to dispaly debug messages
+
+    Returns:
+        None
+
     """
-    Launch a given test.
+    # List of processes to kill on each daos server/client if active
+    daos_process_names = [
+        "orterun",
+        "orted",
+        "daos_server",
+        "daos_io_server",
+        "aexpect-helper-2.7",
+        "avocado",
+        "aexpect-helper-",
+        "daos_agent",
+    ]
+
+    # List of files to remove on each daos server/client
+    daos_files = {
+        "server_log": "/tmp/server.log",
+        "agent_log": "/tmp/daos_agent.log",
+        "control_log": "/tmp/daos_control.log",
+        "sockets_dir": "/tmp/daos_sockets",
+    }
+
+    # Replace any default log files/dirs with yaml definitions
+    yaml_data = get_yaml_data("./data/daos_server_baseline.yaml")
+    display("DAOS server yaml data: {}".format(yaml_data), debug)
+    if isinstance(yaml_data, dict):
+        for key, value in yaml_data.items():
+            if isinstance(value, dict):
+                for key2, value2 in yaml_data[key].items():
+                    if key2 == "log_file":
+                        daos_files["server_log"] = value2
+            elif key == "control_log_file":
+                daos_files["control_log"] = value
+            elif key == "sockets_dir":
+                daos_files["sockets_dir"] = value
+
+    clean_commands = [
+        "killall -9 {}".format(" ".join(daos_process_names)),
+        "rm -rf /mnt/daos/*",
+    ]
+    clean_commands.extend(
+        ["rm -rf {}".format(log) for log in daos_files.values()])
+    for machine in machines:
+        if len(machine) > 0:
+            display("Cleaning test environment on {}".format(machine))
+            output = get_output("ssh {} \"{}\"".format(
+                machine, "; ".join(clean_commands)))
+            display("  Output: {}".format(output), debug)
```

Comment:
Does all of this cleanup need to be done on the test-runner node (i.e. the node running this code) also, which is a daos client (for now)?

Reply:
I don't believe so, but I could be wrong.  There is no guarantee that current tests will run a client/agent on the test-runner.

Reply:
Keep in mind though that unless clients are explicitly defined for a test, the test-runner (i.e. where `launch.py`->`avocado` is being executed) **is** the client and I believe will create a _daos.log_ (maybe not actually named that though -- maybe `client.log` perhaps), which should be cleaned up also.

Another thought though is to use the avocado _workdir_ and let avocado handle it.